File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    r,cl=1,{}
    # 옷 종류와 의상 이름 리스트의 딕셔너리 생성
    # c는 의상 이름, t는 의상 종류
    for c,t in clothes:
        if cl.get(t)==None:
            cl[t]=[c]
        else:
            cl[t]+=[c]
    # 옷 종류 길이를 각각 곱해줍니다.
    for t in cl:
        r*=len(cl[t])+1
    # 결과값에서 1을 빼주어 조합의 수를 구해줍니다.
    return r-1

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int r=1;
    map<string,vector<string>> cl;
    for(int i=0;i<clothes.size();i++) {
        cl[clothes[i][1]].push_back(clothes[i][0]);
    }
    map<string,vector<string>>::iterator itr;
    for(itr=cl.begin();itr!=cl.end();itr++) {
        r*=cl[itr->first].size()+1;
    }
    return r-1;
}